File: LinkedLists/InsertingNode/python/code.py
class Node:

#function to initialize the node object
	def __init__(self, data):
		self.data = data
		self.next = None


class LinkedList:

	def __init__(self):
		self.head=None


	def push(self, newData):
		newNode = Node(newData)
		newNode.next=self.head
		self.head=newNode

	def insertAfter(self, prevNode, newData):
		if prevNode is None:
			return

		newNode = Node(newData)
		newNode.next=prevNode.next
		prevNode.next=newNode

	def append(self, newData):
		newNode = Node(newData)

		if self.head is None:
			self.head=newNode
			return

		last = self.head

		while(last.next):
			last=last.next

		last.next=newNode

	def printList(self):
		temp=self.head
		while(temp):
			print(temp.data, end=" ")
			temp=temp.next


if __name__ =='__main__':

		llist = LinkedList()

		llist.append(1)
		llist.append(2)
		llist.append(3)
		llist.append(4)
		llist.append(5)
		llist.push(2)

		llist.printList()

File: LinkedLists/InsertingNode/java/code.java
class LinkedList {


		Node head;

		class Node {

			int data;
			Node next;

			Node(int val) {
				data=val;
				next=null;
			}
		}

		public void push(int newData) {
			Node newNode = new Node(newData);
			newNode.next=head;
			head=newNode;
		}

		public void insertAfter(Node prevNode, int newData) {
			if(prevNode==null) {
				return;
			}

			Node newNode = new Node(newData);
			newNode.next = prevNode.next;
			prevNode.next = newNode;
		}

		public void append(int newData) {
			Node newNode = new Node(newData);

			if(head==null) {
				head = newNode;
				return;
			}

			Node last = head;
			while(last.next!=null) {
				last=last.next;
			}
			last.next=newNode;
			return;
		}

		public void printList() {
			Node temp = head;
			while(temp!=null) {
				System.out.print(temp.data+" ");
				temp=temp.next;
			}
		}	

	

	public static void main(String[] args) {
		
		LinkedList list = new LinkedList();

		list.append(1);
		list.append(2);
		list.append(3);
		list.append(4);
		list.append(5);


		list.printList();

	}
}